File: src/main/java/TicTacToe/TicTacToe.cs
using System;

class TicTacToe
{
    static void Main()
    {
        char[,] board = new char[3, 3]
        {
            { ' ', ' ', ' ' },
            { ' ', ' ', ' ' },
            { ' ', ' ', ' ' }
        };

        char currentPlayer = 'X';
        bool gameWon = false;

        while (!gameWon)
        {
            PrintBoard(board);

            Console.WriteLine($"Игрок {currentPlayer}, введите строку (0, 1, 2): ");
            int row = int.Parse(Console.ReadLine());
            Console.WriteLine($"Игрок {currentPlayer}, введите столбец (0, 1, 2): ");
            int col = int.Parse(Console.ReadLine());

            if (row >= 0 && row < 3 && col >= 0 && col < 3 && board[row, col] == ' ')
            {
                board[row, col] = currentPlayer;
                gameWon = CheckWin(board, currentPlayer);

                if (gameWon)
                {
                    PrintBoard(board);
                    Console.WriteLine($"Игрок {currentPlayer} выиграл!");
                }
                else
                {
                    currentPlayer = (currentPlayer == 'X') ? 'O' : 'X';
                }
            }
            else
            {
                Console.WriteLine("Некорректный ход. Попробуйте еще раз.");
            }
        }
    }

    static void PrintBoard(char[,] board)
    {
        for (int i = 0; i < 3; i++)
        {
            for (int j = 0; j < 3; j++)
            {
                Console.Write(board[i, j]);
                if (j < 2)
                {
                    Console.Write(" | ");
                }
            }
            Console.WriteLine();
            if (i < 2)
            {
                Console.WriteLine("---------");
            }
        }
    }

    static bool CheckWin(char[,] board, char player)
    {
        for (int i = 0; i < 3; i++)
        {
            if (board[i, 0] == player && board[i, 1] == player && board[i, 2] == player)
            {
                return true;
            }
            if (board[0, i] == player && board[1, i] == player && board[2, i] == player)
            {
                return true;
            }
        }
        if (board[0, 0] == player && board[1, 1] == player && board[2, 2] == player)
        {
            return true;
        }
        if (board[0, 2] == player && board[1, 1] == player && board[2, 0] == player)
        {
            return true;
        }
        return false;
    }
}

File: src/main/java/TicTacToe/TicTacToe.java
package TicTacToe;

public class TicTacToe {
    public static void main(String[] args) {
        // Инициализация переменной gameWon для отслеживания завершения игры.
        boolean gameWon = false;

        // Инициализация двумерного массива для игрового поля 3x3.
        char[][] board = {
                {' ', ' ', ' '},
                {' ', ' ', ' '},
                {' ', ' ', ' '}
        };

        // Начало игрового цикла, который продолжается, пока игра не завершена.
        while (!gameWon) {
            printBoard(board);
            gameWon = true;
        }
    }

    // Метод для отображения игрового поля в консоли.
    public static void printBoard(char[][] board) {
        System.out.println(" | 0 | 1 | 2");
        System.out.println("------------");
        for (int i = 0; i < board.length; i++) {
            System.out.print(i + "| ");
            for (int j = 0; j < board[1].length; j++) {
                // Вывод символа из ячейки игрового поля.
                System.out.print(board[i][j]);
                if (j < 2) {
                    // Разделитель между столбцами
                    System.out.print(" | ");
                }
            }
            // Переход на новую строку после каждой строки игрового поля.
            System.out.println();
            if (i < 2 ) {
                // Горизонтальная линия-разделитель между строками игрового поля.
                System.out.println("------------");
            }
        }
    }
}
